Compute overall dice score

AjeyPaiK opened the pull request:
The PR makes changes:
1. To the way model predictions are accessed in Lightning 2.0
2. To the way how overall dice is computed during validation
3. Adds files to gitignore so that some private files may be maintained locally.

jonasteuwen commented on ahcore/lit_module.py:
```diff
@@ -303,9 +303,9 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
         return gathered_preds
 
-    def on_predict_epoch_end(self, results) -> None:
+    def on_predict_epoch_end(self):
         """Call all the inference trackers to update"""
```
Should do -> None

AjeyPaiK replied:
Yes, done

jonasteuwen commented on ahcore/trackers.py:
```diff
@@ -36,12 +43,12 @@ def __call__(self, predictions: list[list[Tensor]], metadata: dict[str, Any], *a
-        compression: str | None = "jpeg",
-        quality: int | None = 100,
-        is_mask: bool = True,
+            self,
+            save_dir: Union[str, Path],
+            pyramid: bool = False,
+            compression: str | None = "jpeg",
+            quality: int | None = 100,
+            is_mask: bool = True,
     ):
```
What happened?

jonasteuwen reviewed (commented):
Can you explain a bit what you precisely changed in each file?

AjeyPaiK commented:
Changes:

1. Lightning has changed the way it handles prediction outputs. Attached the screenshot below. Earlier, the model outputs were automatically consumed by `on_predict_epoch_end()` but this behavior has changed. Now, we access them through the trainer object as shown below on the right.
![image](https://user-images.githubusercontent.com/26798611/236160905-f86a9b94-1087-4e20-9b8e-ee1d671464b4.png)
So, we make this change in [lit_module](https://github.com/NKI-AI/ahcore/pull/19/files#diff-7503b04ebd2fd6defdbb818002acf81e2ec020d0edb62137f7de35229d56e470), 

2. For my purposes, I need to maintain additional functionalities like a tracker that can measure tumor-stroma ratio during inference which doesn't align with general ahcore functionalities. So, I added such files to gitignore. This will allow me to have them in my local repo whereas the public code neednt have it. This is reflected [here](https://github.com/NKI-AI/ahcore/pull/19/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947).

3. One main change that this PR makes is it adds a flag to the way we compute dice scores over the WSIs. Instead of averaging individual dice scores over the total images, we now aggregate true_positives and cardinalities across the whole validation dataset and treat them as though they originate from a single image. Thus avoiding the need to average while providing more stable DIce estimates. This is reflected [here](https://github.com/NKI-AI/ahcore/pull/19/files#diff-c94e1ef8a11cccfb65a65f64b32c0fbfe082571843caf2c6c9a49953a38880e8).

4. Finally, I have refactored the trackers.py a little bit. I moved the tile iterator out of the Tracker class since it was not dependent on the class variables to function.
